packages/flutter/lib/src/cupertino/date_picker.dart

```diff
@@ -1048,18 +1094,28 @@ class _CupertinoDatePickerDateTimeState extends State<CupertinoDatePicker> {
 
     for (int i = 0; i < columnWidths.length; i++) {
       double offAxisFraction = 0.0;
-      Widget selectionOverlay = _centerSelectionOverlay;
+      Widget? selectionOverlay = _centerSelectionOverlay;
+
+      if (widget.selectionOverlayBuilder != null) {
+        selectionOverlay =
+            widget.selectionOverlayBuilder!(context, i, columnWidths.length);
+      } else {
+        if (i == 0) {
+          selectionOverlay = _startSelectionOverlay;
+        } else if (i == columnWidths.length - 1) {
+          selectionOverlay = _endSelectionOverlay;
+        }
+      }
```

Comment:
Nit: You could do this as a `switch` instead. Not sure if it makes it any cleaner though, up to you.

Also I see that this logic is repeated 3 times. I don't see an easy way to deduplicate it though, you?

Reply:
Sorry @justinmc I can't think of anything to make this more cleaner.